code/lib/types/src/modules/composedStory.ts

```diff
@@ -30,40 +39,60 @@ export type ComposedStoryPlayFn<TRenderer extends Renderer = Renderer, TArgs = A
   context: ComposedStoryPlayContext<TRenderer, TArgs>
 ) => Promise<void> | void;
 
-export type PreparedStoryFn<TRenderer extends Renderer = Renderer, TArgs = Args> = AnnotatedStoryFn<
-  TRenderer,
-  TArgs
-> & { play: ComposedStoryPlayFn<TRenderer, TArgs>; args: TArgs; id: StoryId };
+/**
+ * A loosely annotated story function, used internally by composeStory
+ */
+export type LooselyAnnotatedStoryFn<TRenderer extends Renderer = Renderer, TArgs = Args> = (
```

Comment:
This type is very similar to `PartialStoryFn`:

https://github.com/ComponentDriven/csf/blob/4c735fea4f0c9605b93497238303cb4ab9304727/src/story.ts#L160-L162

I'm not sure if `PartialStoryFn` exactly expresses what that is (sort of like a "bound story function" where it is bound with a context), but this is very similar with a slightly different API so the name could reflect that.

I realize it's an internal API so probably not a huge deal.

Reply:
Yeah, maybe PartialStoryFn is what we need here, as it should also be Partial<TArgs> right?

Reply:
The types might be similar, but they don't match in the function signature. Currently, you use the jsx format for React:
```jsx
<MyComposedStory primary={false} />
```

But for Vue, for instance, you use the function notation:
```js
MyComposedStory({ primary: false })
```

So using the PartialStoryFn type, the signature (and internal implementation) would need to change, which would be a breaking change:
```js
MyComposedStory({ args: { primary: false } })
```

I guess it's debatable; if we make this change we could introduce a way for users to set globals anytime they render the story, in comparison to what is is now:
```tsx
const MyComposedStory = composeStories(myStories.Primary, { globalTypes: { ... } })
MyComposedStory({ primary: false })

versus:

const MyComposedStory = composeStories(myStories.Primary, { globalTypes: { ... } })
MyComposedStory({ args: { primary: false } })
MyComposedStory({ args: { primary: false }, globals: { ... }})
```

Reply:
thoughts @tmeasday @kasperpeulen ?

Reply:
Sorry @yannbf if I put you on the wrong track here. I didn't necessarily mean we needed to reuse `PartialStoryFn`, just that the purpose was very similar so having a similar name would make it less confusing, otherwise someone might be like - what does loosely annotated mean? / what does partial mean? / obviously they are very different concepts, right?

So maybe this should be a `PartialArgsStoryFn`?

Reply:
Got it! I agree and made the change. Please recheck the PR!